src/index.ts

```diff
@@ -17,24 +18,42 @@ import {
 
 type AuthTestFunc = (req: express.Request) => boolean;
 
+class MetricsGathererError extends Error {
```

Comment:
We have a [typed-error](http://github.com/balena-io-modules/typed-error) module you can use here.

Reply:
ah neat, I'll integrate that